src/control/server/harness.go

```diff
@@ -200,6 +200,11 @@ func (h *IOServerHarness) Start(parent context.Context) error {
 		}
 	}
 
+	// Mark the harness as started to block any format/update requests.
+	h.Lock()
+	h.started = true
```

Comment:
I'm a little curious how this is used elsewhere. Is the "started" member only used to see if format can occur? Would it make sense to change this bool to a state enum? I.e. allow format only in the "waiting for format" state, and then have a "starting" state here, and a "ready" state when the IO servers are up and running.

Reply:
At the moment it's just a simple guard to ensure that we reject certain requests (format, firmware update, etc) if the harness has proceeded to at least trying to start the instances. See the RPC handlers in ctl_storage_rpc.go for more details on usage.

I'm not opposed to the idea of making it an enum, but at the moment not really sure what problem that would solve. Looking at this again, though, I do think it would be best to move this up to the top of the method to eliminate any race between starting the instances and setting this bool. It'd be a small window, but it's possible that a stray RPC could come in during this time and result in undefined behavior.

Reply:
Works for me. It was more a question than anything - if it's used only for rejecting RPCs after the IO servers have started, I don't see the use of changing it either.